### ISSUE TITLE
How can I implement row/column normalization of a sparse matrix using softmax or other functions?

### ISSUE THREAD

**Issue by RenShijie** [pyg-team/pytorch_geometric#1074]
I have a large m*m sparse symmetric matrix X. I would like to normalize each row of X, so that each element in the same row turns into a softmax manner exp(X(i,j))/sum(exp(X(i,:))). Instead of transforming sparse tensor into a dense one or using loops, I wonder if there is a fast implementation to operate on all rows/cols and perform summation/product/division/exponentation for a sparse matrix, especially when the element numbers in rows/cols are not always equal. I thought about combining loop and indexing, but that's too slow for a large matrix. So...Is there any sparse solution for this issue?

**Comment by rusty1s** [comment]
You can extend the [`softmax` operator of PyG](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py) to this use-case.

**Comment by RenShijie** [comment]
Thanks for your help, that really works! ^_^
